Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -2475,19 +2485,28 @@ else if (sourceProperty.getScope() != null) {
             Query query = of.createQuery().withSource(querySource).withWhere(criteria);
             result = query;
 
-            if (indexerEnd < targetMap.length()) {
+            if (indexerEnd + 1 < targetMap.length()) {
                 // There are additional paths following the indexer, apply them
                 String targetPath = targetMap.substring(indexerEnd + 1);
                 if (targetPath.startsWith(".")) {
                     targetPath = targetPath.substring(1);
                 }
 
+                if (!targetPath.isEmpty()) {
+                    query.setReturn(of.createReturnClause()
+                            .withDistinct(false).withExpression(of.createProperty()
```

Comment:
Why is this distinct "false" as opposed to true?

Reply:
Because the source data may have duplicates and they would be expected to be returned. It may be an extension, for example, and we wouldn't be justified in applying a distinct. There may be cases where we would be justified, but we're not detecting those currently.